### PR TITLE
enhance(library): update Duration() return values

### PR DESCRIPTION
**Pull request by jbrockopp**
Based off feedback from https://github.com/go-vela/types/pull/222#issuecomment-1003031343

This updates the value returned from `Duration()` for resources in the `github.com/go-vela/types/library` package.

The following resources have this function:

* [build](https://github.com/go-vela/types/blob/c491e96440a50566b68980de606748ec7c469acd/library/build.go#L51-L70)
* [service](https://github.com/go-vela/types/blob/c491e96440a50566b68980de606748ec7c469acd/library/service.go#L36-L55)
* [step](https://github.com/go-vela/types/blob/c491e96440a50566b68980de606748ec7c469acd/library/step.go#L38-L57)

Previously, we were returning `0s` if the `started` or `finished` timestamps weren't set for the resource.

Now, If a resource doesn't have a `started` timestamp, then an indeterminate value of `...` is returned.

Otherwise, if a resource doesn't have a `finished` timestamp, then the time since the `started` timestamp is returned.

Otherwise, we return the `finished` timestamp subtracted by the `started` timestamp.

### CODE REVIEW

File: library/build.go [hunk 1]
```diff
@@ -51,21 +51,30 @@
 // Duration calculates and returns the total amount of
 // time the build ran for in a human-readable format.
 func (b *Build) Duration() string {
-	// check if the build doesn't have a started or finished timestamp
-	if b.GetStarted() == 0 || b.GetFinished() == 0 {
-		// return zero value for time.Duration (0s)
-		return new(time.Duration).String()
+	// check if the build doesn't have a started timestamp
+	if b.GetStarted() == 0 {
+		return "..."
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply library/build.go:57]: fixed 👍

File: library/service.go [hunk 2]
```diff
@@ -36,21 +36,31 @@ type Service struct {
 // Duration calculates and returns the total amount of
 // time the service ran for in a human-readable format.
 func (s *Service) Duration() string {
-	// check if the service doesn't have a started or finished timestamp
-	if s.GetStarted() == 0 || s.GetFinished() == 0 {
-		// return zero value for time.Duration (0s)
-		return new(time.Duration).String()
+	// check if the service doesn't have a started timestamp
+	if s.GetStarted() == 0 {
+		// nolint: goconst // ignore making a constant
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply library/service.go]: fixed 👍

File: library/step.go [hunk 3]
```diff
@@ -38,21 +38,31 @@ type Step struct {
 // Duration calculates and returns the total amount of
 // time the step ran for in a human-readable format.
 func (s *Step) Duration() string {
-	// check if the step doesn't have a started or finished timestamp
-	if s.GetStarted() == 0 || s.GetFinished() == 0 {
-		// return zero value for time.Duration (0s)
-		return new(time.Duration).String()
+	// check if the step doesn't have a started timestamp
+	if s.GetStarted() == 0 {
+		// nolint: goconst // ignore making a constant
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply library/step.go]: fixed 👍

**Review by dtanner — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]
lgtm

**Review by JordanSussman — APPROVED** [review]

**Pull request merged**